### PR TITLE
Collaborator List Permission Retention

### PR DESCRIPTION
**Pull request by dennisroberts71**
This change allows users to grant the permissions that had been granted to a collaborator list directly to be granted to members that are removed from a collaborator list or to all collaborator list members if the collaborator list is being deleted.

Update: I had some other minor changes to the subject search endpoint that were finished shortly after I submitted this PR. Rather than create a new branch and a new PR, I just included these changes with this PR.

### CODE REVIEW

File: src/terrain/services/collaborator_lists.clj [hunk 1]
```diff
@@ -16,16 +16,37 @@
...
 (defn get-collaborator-list-members [{user :shortUsername} name]
   (ipg/get-collaborator-list-members user name))
 
 (defn add-collaborator-list-members [{user :shortUsername} name {:keys [members]}]
   (ipg/add-collaborator-list-members user name members))
 
-(defn remove-collaborator-list-members [{user :shortUsername} name {:keys [members]}]
-  (ipg/remove-collaborator-list-members user name members))
+(defn- copy-collaborator-list-permissions [user group-id {:keys [results]}]
+  (when-let [subjects (seq (for [result results :when (:success result)] (perms-subject-for result)))]
```
Comment (psarando) [inline src/terrain/services/collaborator_lists.clj]: 👍

File: project.clj [hunk 2]
```diff
@@ -40,7 +40,7 @@
                  [org.cyverse/common-cli "2.8.1"]
                  [org.cyverse/kameleon "3.0.1"]
                  [org.cyverse/heuristomancer "2.8.0"]
-                 [org.cyverse/permissions-client "2.8.0"]
+                 [org.cyverse/permissions-client "2.8.1-SNAPSHOT"]
```
Comment (psarando) [inline project.clj]: Recommend bumping to release version before merging.

Comment (dennisroberts71) [inline project.clj]: Will do.

**Comment by dennisroberts71** [comment]
Thanks for the review!

**Pull request merged**